Sprint 13

maxim-papier opened the pull request.

maxim-papier committed:
Merge pull request #19 from maxim-papier/sprint_12

Update 13 from 12

maxim-papier committed:
Set roadmap for ImagesList refactoring

maxim-papier committed:
Can't get ImageListService

maxim-papier committed:
Fixed presenter init issue

maxim-papier committed:
+ ImageListVC is refactored

maxim-papier committed:
Re-refactoring of ProfileViewVC and his presenter

maxim-papier committed:
Refactoring

maxim-papier committed:
+ Tests for ProfileVC <-> ProfilePresenter <-> LogoutHelper()

maxim-papier committed:
+ CoreAnimation gradient

maxim-papier committed:
+ Set presenter/view relations between ImagesListVC and ImagesListPresenter

maxim-papier committed:
+ Tests are ready

maxim-papier committed:
Stuck with infinity gradient animation on profile page

maxim-papier committed:
Clean up and ready for merge

maxim-papier committed:
Merge pull request #22 from maxim-papier/sprint_13_presenter

Sprint 13 presenter

IBDesignable commented on Gallery/Scenes/Profile/ProfilePresenter.swift:
```diff
@@ -0,0 +1,68 @@
+
+final class ProfilePresenter: ProfilePresenterProtocol {
+    
+    var view: ProfileViewControllerProtocol?
```
weak

IBDesignable commented on Gallery/Scenes/Profile/ProfilePresenter.swift:
```diff
@@ -0,0 +1,68 @@
+    
+    var view: ProfileViewControllerProtocol?
+    
+    let profileService = ProfileService.shared
+    let profileImageService = ProfileImageService.shared
```
Они нужны для тестов? давай все что не нужно не закрыто ProfilePresenterProtocol закроем приватными модификаторами

IBDesignable commented:
В пул реквест не попал код из чеклиста. Будь пожалуйста аккуратнее с комитами

maxim-papier commented on Gallery/Helpers/GradientKFIndicator.swift:
```diff
@@ -0,0 +1,23 @@
+//import Kingfisher
+//import UIKit
+//
+//final class GradientKFIndicator: Indicator {
+//    var view: UIView { gradientView }
+//
+//    var gradientView: GradientView = {
+//        let gradientView = GradientView()
+//        gradientView.configureForLoader()
+//
+//        return gradientView
+//    }()
+//
+//    func startAnimatingView() {
+//        gradientView.animateLoader()
+//        view.isHidden = false
+//    }
+//
+//    func stopAnimatingView() {
+//        gradientView.stopAnimationLoader()
+//        view.isHidden = true
+//    }
```
Ай, стыдоба какая.

maxim-papier commented on Gallery/Helpers/GradientView.swift:
```diff
@@ -0,0 +1,70 @@
+//import UIKit
+//
+//final class GradientView: UIView {
+//
+//    var gradient = CAGradientLayer()
+//
+//    override init(frame: CGRect) {
+//        super.init(frame: frame)
+//        gradient.frame = self.bounds
```
Всё понял. Я экспериментировал и удалил файлы, но, видимо, как-то неправильно — в Xcode мне  их не видно. 

Удалил вручную через Finder.

maxim-papier commented on GalleryTests/Profile/ProfileViewTest.swift:
```diff
@@ -0,0 +1,97 @@
+final class ProfileViewTest: XCTestCase {
+    
+    
+    // Let's check if the loginName on the ProfileVC
+    // are updated when the presenter method updateProfile()
+    // is called
+        
+    func testPresenterUpdatesLoginNameLabelVC() {
+
+        ///given
```
Спасибо. Очень интересно про документацию. Исправил.

maxim-papier committed:
# Minor fixes for code review tasks

maxim-papier committed:
#2 Minor fixes for code review tasks

maxim-papier committed:
Merge pull request #26 from maxim-papier/sprint_13_presenter

Sprint 13 presenter

maxim-papier committed:
#3 Major fixe for code review tasks:

- Move UI-responsibility from the presenter to the controller in ImagesList

maxim-papier committed:
#3 Major fixes for code review tasks:
- Moving liking from presenter to controller
- Moving setting right heart icon the liked to CellImageService

maxim-papier commented on Gallery/Scenes/ImagesList/ImagesListPresenter.swift:
```diff
@@ -0,0 +1,117 @@
+import UIKit
```
Совсем что-то не вышло избавиться. updateTableViewAnimated() метод никак не растащу между презентером и вьюконтроллером.

IBDesignable replied:
Окей, это действительно может быть не тривиально сейчас, поэтому я и написала стремиться. 

Оставь как есть

maxim-papier commented on Gallery/Scenes/ImagesList/ImagesListPresenter.swift:
```diff
@@ -0,0 +1,117 @@
+    
+    init(service: ImageListService) {
+        self.service = service
+        self.dateFormatter = DateFormatter()
+        self.dateFormatter.dateStyle = .long
+        self.dateFormatter.timeStyle = .none
```
Отложил себе в issues

maxim-papier committed:
# Code review  №1 fixes are done

maxim-papier commented:
Добрый день, Александра. Спасибо за ваши замечания. Приношу свои извинения за неаккуратную работу. Я очень спешил. Мне поставили дедлайн до конца понедельника и я сделал пулрееквест прямо впритык. 

Я исправил то, что мог. Посмотрите, пожалуйста, требуется ли ещё доработки?

IBDesignable commented on Gallery/Scenes/TabBarController.swift:
```diff
@@ -3,33 +3,29 @@ import UIKit
-        let imageListService = ImageListService()
-        
-        imagesListVC.presenter = ImagesListPresenter(service: imageListService)
+        let service = ImageListService()
+        let presenter1 = ImagesListPresenter(service: service)
+        vc1.presenter = presenter1
+        presenter1.view = vc1
```
Не приветствуются именования 1 и 2, ничего страшного не произойдет если ты напишешь полное название imageListService, imageListPresenter

Но если правда лень – тогда хотя бы выноси в отдельные функции – assembleImageListModule -> UIViewController 

На самом деле подход с фабричными методами близко к продакшену будет :) asseble/build/create глаголы которые чаще всего в таком коде можно встретить

maxim-papier replied:
Я не ленился, а пытался сократить —очень не люблю длиннющие имена. :) Больше так не буду. Спасибо.

IBDesignable replied:
мне тоже так по первости казалось, а потом оказалось что лучше длинные, чтобы было понятно из любой точки кода
